tsl/src/fdw/relinfo.c

```diff
@@ -257,6 +257,14 @@ estimate_tuples_and_pages_using_shared_buffers(PlannerInfo *root, Hypertable *ht
 static void
 estimate_chunk_size(PlannerInfo *root, RelOptInfo *chunk_rel)
 {
+	/*
+	 * In any case, cache the chunk info for this chunk.
+	 */
+	RangeTblEntry *chunk_rte = planner_rt_fetch(chunk_rel->relid, root);
+	TsFdwRelInfo *chunk_private = fdw_relinfo_get(chunk_rel);
+	Assert(!chunk_private->chunk);
+	chunk_private->chunk = ts_chunk_get_by_relid(chunk_rte->relid, true /* fail_if_not_found */);
```

Comment:
Is this function is a good place to save chunk information, give it's name? Also wondering if moving chunk search out of `hypertable_cache_pin()` has any possible side effects

Reply:
It doesn't use the hypertable cache, so should be OK.

Not a good place, I'd want to do the chunk lookup as a separate loop. All places are all kind of bad because we only have some functions called by postgres at some moments... I'll try to find someplace better.

Reply:
The best place seems to be `ts_plan_expand_hypertable`, but it's not called for UPDATEs, they use `expand_inherited_rtentry` instead. I think I'll write this in a comment and keep the caching here. When we have chunk exclusion for UPDATEs maybe there will be a more convenient place for this.